lib/pure/strformat.nim

```diff
@@ -552,8 +552,22 @@ proc strformatImpl(pattern: NimNode; openChar, closeChar: char): NimNode =
 
         var subexpr = ""
         while i < f.len and f[i] != closeChar and f[i] != ':':
-          subexpr.add f[i]
-          inc i
+          if f[i] == '=':
+            let start = i
+            inc i
+            i += f.skipWhitespace(i)
+            if f[i] == closeChar or f[i] == ':':
+              try:
+                discard parseExpr(subexpr)
+                result.add newCall(bindSym"add", res, newLit(subexpr & f[start ..< i]))
+              except ValueError:
```

Comment:
That looks dangerous. Secretly misinterpreting what somebody might have intented.

Reply:
I think fmt's safety is not much difference with the previous? The previous code used `parseExpr` too.

Reply:
Can you remove the ``except ValueError: ...`` section please?